<a href="https://colab.research.google.com/github/jcesarmph-ui/N1---2-Bimestre---Individual---An-lise-Explorat-ria-de-Dados-EDA-/blob/main/Aprofundamento_Estatistico_JulioCesar_ipynb_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Identificação do Problema

Durante a análise exploratória, foi observado que máquinas com maior torque tendem a apresentar mais falhas.

O objetivo agora é verificar se essa relação é estatisticamente significativa  ou não.

In [5]:
import pandas as pd
import numpy as np

from scipy.stats import shapiro
from scipy.stats import mannwhitneyu

In [6]:
import pandas as pd

df = pd.read_csv('ai4i2020.csv')

df.head()

,UDI,Product ID,Type,Air temperature [K],Process temperature [K],Rotational speed [rpm],Torque [Nm],Tool wear [min],Machine failure,TWF,HDF,PWF,OSF,RNF
0,1,M14860,M,298.1,308.6,1551,42.8,0,0,0,0,0,0,0
1,2,L47181,L,298.2,308.7,1408,46.3,3,0,0,0,0,0,0
2,3,L47182,L,298.1,308.5,1498,49.4,5,0,0,0,0,0,0
3,4,L47183,L,298.2,308.6,1433,39.5,7,0,0,0,0,0,0
4,5,L47184,L,298.2,308.7,1408,40.0,9,0,0,0,0,0,0


In [7]:
# Separando os grupos

falha = df[df['Machine failure'] == 1]['Torque [Nm]']
sem_falha = df[df['Machine failure'] == 0]['Torque [Nm]']

##Teste de normalidade (Shapiro-Wilk)

Deseja-se verificar se existe diferença significativa no torque entre máquinas que falham e máquinas que não falham.

In [8]:
stat_falha, p_falha = shapiro(falha)
stat_sem, p_sem = shapiro(sem_falha)

print("Falha p-value:", p_falha)
print("Sem falha p-value:", p_sem)

Falha p-value: 1.4250043105130285e-15
Sem falha p-value: 8.100950106412522e-09


/usr/local/lib/python3.12/dist-packages/scipy/stats/_axis_nan_policy.py:579: UserWarning: scipy.stats.shapiro: For N > 5000, computed p-value may not be accurate. Current N is 9661.
  res = hypotest_fun_out(*samples, **kwds)


###RESULTADOS

Os dados foram avaliados utilizando o teste de Shapiro-Wilk.
Se p > 0.05 → normal
Se p < 0.05 → não normal

Os resultados foram: Falha p-value: 1.42e-15
Sem falha p-value: 8.10e-09
Ou seja, valores menores que 0.05, portanto, os dados não seguem distribuição normal, sendo necessário utilizar testes não paramétricos.

#Hipóteses (Teste de Mann-Whitney)

H0 (Hipótese Nula):
Não existe diferença significativa no torque entre máquinas que falham e que não falham.

H1 (Hipótese Alternativa):
Existe diferença significativa no torque entre máquinas que falham e que não falham.

In [9]:
# Teste de Mann-Whitney

stat, p = mannwhitneyu(falha, sem_falha)

print("p-value:", p)

p-value: 2.262232331232534e-64


###Resultados

Foi aplicado o teste de Mann-Whitney para comparar o torque entre máquinas que falham e que não falham.

O p-valor obtido foi: p-value = 2.26e-64

Como o p-valor é muito menor que 0,05, rejeita-se a hipótese nula (H0).

Isso indica que existe uma diferença estatisticamente significativa entre os dois grupos, ou seja, o torque das máquinas com falha é significativamente diferente do torque das máquinas sem falha.

In [10]:
# Cálculo do D de Cohen

import numpy as np

mean1 = np.mean(falha)
mean2 = np.mean(sem_falha)

std1 = np.std(falha)
std2 = np.std(sem_falha)

d = (mean1 - mean2) / np.sqrt((std1**2 + std2**2) / 2)

print("Cohen's d:", d)

Cohen's d: 0.788736963251303


##Tamanho do Efeito

O tamanho do efeito foi calculado utilizando o d de Cohen, que mede a magnitude da diferença entre os grupos.

Valores de referência:
0.2 → efeito pequeno  
0.5 → efeito médio  
0.8 ou mais → efeito grande  

O D de Cohen calculado foi: 0.79

Esse valor indica um efeito de magnitude média a alta, próximo de um efeito grande. Isso significa que, além da diferença ser estatisticamente significativa (confirmada pelo p-valor), ela também possui relevância prática no contexto do problema. Portanto, o torque apresenta uma diferença consistente entre máquinas que falham e que não falham, sendo um fator importante para a análise e predição de falhas industriais.

#CONCLUSÃO CIENTÍFICA

Com base nos testes estatísticos realizados, a hipótese nula (H0) foi rejeitada, visto que o p-valor obtido (2.26e-64) é extremamente inferior ao nível de significância adotado (0,05). Isso indica que a diferença observada no torque entre máquinas que falham e que não falham não ocorreu ao acaso, sendo estatisticamente significativa.

Além disso, o tamanho do efeito calculado (Cohen's d = 0.79) revelou uma magnitude média a alta, demonstrando que essa diferença possui relevância prática no contexto industrial.

Do ponto de vista do problema de negócio, esse resultado confirma que o torque é um fator crítico na ocorrência de falhas em equipamentos. Isso implica que o monitoramento dessa variável pode ser utilizado como um indicador confiável para antecipar falhas, contribuindo diretamente para estratégias de manutenção preditiva e redução de paradas não planejadas. Sua relevância estatística e prática aumenta a capacidade preditiva do sistema, auxiliando na construção de modelos mais precisos e eficientes.